Design Issue: validateQueryArgs uses global viper state (not thread-safe)

## Issue Description

The `validateQueryArgs()` function relies on global viper state for configuration, which is not thread-safe. Concurrent calls to viper operations (especially `viper.Reset()`) can cause panics and data races.

## Location

- **File:** `cmd/query.go`
- **Function:** `validateQueryArgs(ctx context.Context, args []string) error`
- **Lines:** 153-178 (multiple viper.IsSet, viper.GetString, viper.GetStringSlice calls)

## Impact

**Severity:** LOW (Current Usage Safe)

**Current State:** Not an issue in practice because:
- Validation happens sequentially in command handler
- No concurrent validation calls in production

**Potential Issues:**
1. **Testing difficulties:** Tests that use viper.Reset() can't run in parallel
2. **Future risk:** If validation ever called concurrently, will panic
3. **Testability:** Hard to test in isolation without global state

## Reproduction

The issue can be demonstrated in tests:

```go
func TestValidateQueryArgs_ConcurrentCalls(t *testing.T) {
    ctx := context.Background()
    var wg sync.WaitGroup
    
    for i := 0; i < 100; i++ {
        wg.Add(1)
        go func() {
            defer wg.Done()
            viper.Reset()  // ← RACE: Concurrent Reset() causes panics
            viper.Set(pconstants.ArgOutput, constants.OutputFormatTable)
            validateQueryArgs(ctx, []string{"SELECT 1"})
        }()
    }
    
    wg.Wait()  // Panics with concurrent viper access
}
```

**Test Output:**
```
panic: concurrent map write
```

## Root Cause

Viper uses global state and is not designed for concurrent access:

```go
func validateQueryArgs(ctx context.Context, args []string) error {
    // All these use global viper state:
    interactiveMode := len(args) == 0
    if viper.IsSet(pconstants.ArgSnapshot) || viper.IsSet(pconstants.ArgShare) {
        // ...
    }
    if len(viper.GetStringSlice(pconstants.ArgExport)) > 0 {
        // ...
    }
    output := viper.GetString(pconstants.ArgOutput)
    // ...
}
```

## Suggested Fixes

**Option 1: Pass configuration as struct (best for testability)**
```go
type QueryConfig struct {
    Snapshot      bool
    Share         bool
    Export        []string
    Output        string
    SnapshotTitle string
    // ... other config fields
}

func validateQueryArgs(ctx context.Context, args []string, cfg *QueryConfig) error {
    interactiveMode := len(args) == 0
    if interactiveMode && (cfg.Snapshot || cfg.Share) {
        return sperr.New("cannot share snapshots in interactive mode")
    }
    // ... no global state access
}

// Caller constructs config from viper once:
cfg := &QueryConfig{
    Snapshot: viper.GetBool(pconstants.ArgSnapshot),
    Share:    viper.GetBool(pconstants.ArgShare),
    Export:   viper.GetStringSlice(pconstants.ArgExport),
    Output:   viper.GetString(pconstants.ArgOutput),
}
err := validateQueryArgs(ctx, args, cfg)
```

**Benefits:**
- ✅ No global state access
- ✅ Thread-safe by design
- ✅ Easy to test (just pass different config structs)
- ✅ Clear what configuration is needed

**Option 2: Protect viper access with mutex (simpler but less ideal)**
```go
var viperMutex sync.RWMutex

func validateQueryArgs(ctx context.Context, args []string) error {
    viperMutex.RLock()
    snapshot := viper.GetBool(pconstants.ArgSnapshot)
    share := viper.GetBool(pconstants.ArgShare)
    // ... read all config
    viperMutex.RUnlock()
    
    // Do validation without holding lock
}
```

**Benefits:**
- ✅ Minimal code changes
- ✅ Thread-safe

**Drawbacks:**
- ❌ Still uses global state
- ❌ Harder to test
- ❌ Must remember to lock everywhere

**Option 3: Use viper sub-instances (if viper supports)**
```go
// Create instance per validation
cfg := viper.Sub("query")
// Use cfg instead of global viper
```

## Test Coverage

This issue is documented in:
- **Test:** `TestValidateQueryArgs_ConcurrentCalls` in `cmd/query_test.go`
- **Status:** Test is skipped with documentation of the issue

## Why This Matters

1. **Testing:** Can't run tests with viper.Reset() in parallel
2. **Future-proofing:** If concurrent validation needed, will break
3. **Best practices:** Global mutable state is an anti-pattern
4. **Testability:** Hard to test in isolation

## Recommendation

**Priority:** Low
- Works correctly in current usage
- Good refactoring opportunity
- Consider as part of larger refactoring initiative
- **Option 1** (pass config struct) is best for long-term maintainability

## Additional Context

- Found during Wave 2 testing quality initiative
- Similar pattern likely exists in other command validation functions
- Part of broader "minimize global state" improvement opportunity
- Related to issue #4705 (global exitCode side effects)

## Related Issues

- #4705 - Global exitCode side effects
- Consider a larger refactoring initiative to reduce global state throughout cmd package

## Labels

`technical-debt`, `refactoring`, `testing`, `thread-safety`, `area:query`

--- a/cmd/query.go
+++ b/cmd/query.go
@@ -30,6 +30,15 @@ var queryTimingMode = constants.QueryTimingModeOff
 // variable used to assign the output mode flag
 var queryOutputMode = constants.QueryOutputModeTable
 
+// queryConfig holds the configuration needed for query validation
+// This avoids concurrent access to global viper state
+type queryConfig struct {
+	snapshot bool
+	share    bool
+	export   []string
+	output   string
+}
+
 func queryCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:              "query",
@@ -93,8 +102,16 @@ func runQueryCmd(cmd *cobra.Command, args []string) {
 		}
 	}()
 
+	// Read configuration from viper once to avoid concurrent access issues
+	cfg := &queryConfig{
+		snapshot: viper.IsSet(pconstants.ArgSnapshot),
+		share:    viper.IsSet(pconstants.ArgShare),
+		export:   viper.GetStringSlice(pconstants.ArgExport),
+		output:   viper.GetString(pconstants.ArgOutput),
+	}
+
 	// validate args
-	err := validateQueryArgs(ctx, args)
+	err := validateQueryArgs(ctx, args, cfg)
 	error_helpers.FailOnError(err)
 
 	// if diagnostic mode is set, print out config and return
@@ -150,13 +167,13 @@ func runQueryCmd(cmd *cobra.Command, args []string) {
 	}
 }
 
-func validateQueryArgs(ctx context.Context, args []string) error {
+func validateQueryArgs(ctx context.Context, args []string, cfg *queryConfig) error {
 	interactiveMode := len(args) == 0
-	if interactiveMode && (viper.IsSet(pconstants.ArgSnapshot) || viper.IsSet(pconstants.ArgShare)) {
+	if interactiveMode && (cfg.snapshot || cfg.share) {
 		exitCode = constants.ExitCodeInsufficientOrWrongInputs
 		return sperr.New("cannot share snapshots in interactive mode")
 	}
-	if interactiveMode && len(viper.GetStringSlice(pconstants.ArgExport)) > 0 {
+	if interactiveMode && len(cfg.export) > 0 {
 		exitCode = constants.ExitCodeInsufficientOrWrongInputs
 		return sperr.New("cannot export query results in interactive mode")
 	}
@@ -168,10 +185,9 @@ func validateQueryArgs(ctx context.Context, args []string) error {
 	}
 
 	validOutputFormats := []string{constants.OutputFormatLine, constants.OutputFormatCSV, constants.OutputFormatTable, constants.OutputFormatJSON, constants.OutputFormatSnapshot, constants.OutputFormatSnapshotShort, constants.OutputFormatNone}
-	output := viper.GetString(pconstants.ArgOutput)
-	if !slices.Contains(validOutputFormats, output) {
+	if !slices.Contains(validOutputFormats, cfg.output) {
 		exitCode = constants.ExitCodeInsufficientOrWrongInputs
-		return sperr.New("invalid output format: '%s', must be one of [%s]", output, strings.Join(validOutputFormats, ", "))
+		return sperr.New("invalid output format: '%s', must be one of [%s]", cfg.output, strings.Join(validOutputFormats, ", "))
 	}
 
 	return nil

--- a/cmd/query_test.go
+++ b/cmd/query_test.go
@@ -7,9 +7,7 @@ import (
 	"sync"
 	"testing"
 
-	"github.com/spf13/viper"
 	"github.com/stretchr/testify/assert"
-	pconstants "github.com/turbot/pipe-fittings/v2/constants"
 	"github.com/turbot/steampipe/v2/pkg/constants"
 )
 
@@ -73,13 +71,17 @@ func TestValidateQueryArgs_ConcurrentCalls(t *testing.T) {
 		go func(iteration int) {
 			defer wg.Done()
 
-			// Each goroutine resets and sets viper config
-			// This simulates concurrent test execution or potential concurrent validation
-			viper.Reset()
-			viper.Set(pconstants.ArgOutput, constants.OutputFormatTable)
+			// Create config struct - this is now thread-safe
+			// Each goroutine has its own config instance
+			cfg := &queryConfig{
+				snapshot: false,
+				share:    false,
+				export:   []string{},
+				output:   constants.OutputFormatTable,
+			}
 
 			// Call validateQueryArgs with a query argument (non-interactive mode)
-			err := validateQueryArgs(ctx, []string{"SELECT 1"})
+			err := validateQueryArgs(ctx, []string{"SELECT 1"}, cfg)
 			if err != nil {
 				errors <- err
 			}
@@ -103,13 +105,16 @@ func TestValidateQueryArgs_ConcurrentCalls(t *testing.T) {
 func TestValidateQueryArgs_InteractiveModeWithSnapshot(t *testing.T) {
 	ctx := context.Background()
 
-	// Setup viper config
-	viper.Reset()
-	viper.Set(pconstants.ArgSnapshot, true)
-	viper.Set(pconstants.ArgOutput, constants.OutputFormatTable)
+	// Setup config with snapshot enabled
+	cfg := &queryConfig{
+		snapshot: true,
+		share:    false,
+		export:   []string{},
+		output:   constants.OutputFormatTable,
+	}
 
 	// Call with no args (interactive mode)
-	err := validateQueryArgs(ctx, []string{})
+	err := validateQueryArgs(ctx, []string{}, cfg)
 
 	// Should return error for snapshot in interactive mode
 	assert.Error(t, err)
@@ -120,13 +125,16 @@ func TestValidateQueryArgs_InteractiveModeWithSnapshot(t *testing.T) {
 func TestValidateQueryArgs_BatchModeWithSnapshot(t *testing.T) {
 	ctx := context.Background()
 
-	// Setup viper config
-	viper.Reset()
-	viper.Set(pconstants.ArgSnapshot, true)
-	viper.Set(pconstants.ArgOutput, constants.OutputFormatTable)
+	// Setup config with snapshot enabled
+	cfg := &queryConfig{
+		snapshot: true,
+		share:    false,
+		export:   []string{},
+		output:   constants.OutputFormatTable,
+	}
 
 	// Call with args (batch mode)
-	err := validateQueryArgs(ctx, []string{"SELECT 1"})
+	err := validateQueryArgs(ctx, []string{"SELECT 1"}, cfg)
 
 	// Should not return error for snapshot in batch mode
 	// (unless there are other validation errors from cmdconfig.ValidateSnapshotArgs)
@@ -141,12 +149,16 @@ func TestValidateQueryArgs_BatchModeWithSnapshot(t *testing.T) {
 func TestValidateQueryArgs_InvalidOutputFormat(t *testing.T) {
 	ctx := context.Background()
 
-	// Setup viper config with invalid output format
-	viper.Reset()
-	viper.Set(pconstants.ArgOutput, "invalid-format")
+	// Setup config with invalid output format
+	cfg := &queryConfig{
+		snapshot: false,
+		share:    false,
+		export:   []string{},
+		output:   "invalid-format",
+	}
 
 	// Call with args (batch mode)
-	err := validateQueryArgs(ctx, []string{"SELECT 1"})
+	err := validateQueryArgs(ctx, []string{"SELECT 1"}, cfg)
 
 	// Should return error for invalid output format
 	assert.Error(t, err)

